cvat/apps/engine/task.py

```diff
@@ -482,10 +482,19 @@ def _create_task_manifest_from_cloud_data(
     sorted_media: List[str],
     manifest: ImageManifestManager,
     dimension: models.DimensionType = models.DimensionType.DIM_2D,
+    *,
+    stop_frame: Optional[int] = None,
 ) -> None:
+    if stop_frame is None:
+        stop_frame = len(sorted_media) - 1
     cloud_storage_instance = db_storage_to_storage_instance(db_storage)
-    content = cloud_storage_instance.bulk_download_to_memory(sorted_media)
-    manifest.link(sources=content, DIM_3D=dimension == models.DimensionType.DIM_3D)
+    content_generator = cloud_storage_instance.bulk_download_to_memory(sorted_media)
+
+    manifest.link(
+        sources=content_generator,
+        DIM_3D=dimension == models.DimensionType.DIM_3D,
+        stop=stop_frame,
+    )
```

Comment:
Did you check if the reverse order of results, returned by the generator, is not a problems here?

Reply:
Why do you think that results will be reversed? Yes, I've checked that after using different sorting methods tasks will be created with the correct, expected file order. I guess we have such tests but need to check.

Reply:
You can see it in [my previous comment](https://github.com/cvat-ai/cvat/pull/7903#discussion_r1615919002) with the small script output - the item numbers are reversed. You can also find it in the CPython implementation [here](https://github.com/python/cpython/blob/48f9d3e3faec5faaa4f7c9849fecd27eae4da213/Lib/concurrent/futures/_base.py#L616-L617).

Reply:
Yes, I also checked the source code before asking you but you can see that after reversing the list with futures, the next future will be the last one from the reversed list that corresponds to the first one from the original list. So I still don't understand..

Reply:
Ok, I investigated it one more time, it seems to be the detail of the list call.

Reply:
@zhiltsov-max, I've added REST API tests to check task creation using cloud data and different sorting methods.